### PR TITLE
Bump Gradle to 8.0 and Gradle plugin to 8.0.1

### PR DESCRIPTION
**Pull request by rmartin16**
## Changes
- Updating Gradle to accommodate updating JDK in beeware/briefcase#1271

BRIEFCASE_REPO: https://github.com/rmartin16/briefcase/
BRIEFCASE_REF: bump-jdk

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by mhsmith** [comment]
See https://github.com/beeware/briefcase/pull/1271#issuecomment-1549724397.

File: {{ cookiecutter.format }}/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-all.zip
```
Comment (mhsmith) [inline {{ cookiecutter.format }}/gradle/wrapper/gradle-wrapper.properties:6]: To maximize compatibility, the Gradle version in this file and the Android Gradle plugin version in build.gradle should be a combination which is generated by the new project wizard of the corresponding Android Studio version. For Android Studio Flamingo, that combination is:
* Gradle 8.0
* Android Gradle plugin 8.0.1

Comment (rmartin16) [reply {{ cookiecutter.format }}/gradle/wrapper/gradle-wrapper.properties]: Good deal; thanks. Bumped Gradle to 8.0 instead of 8.1.1.

**Review by freakboy3742 — APPROVED** [review]
👍

**Pull request merged**